presto-docs/src/main/sphinx/include/PrestoThriftService.thrift

```diff
@@ -318,6 +318,21 @@ service PrestoThriftService {
   PrestoThriftSplitBatch prestoGetSplits(1:  PrestoThriftSchemaTableName schemaTableName, 2:  PrestoThriftNullableColumnSet desiredColumns, 3:  PrestoThriftTupleDomain outputConstraint, 4:  i32 maxSplitCount, 5:  PrestoThriftNullableToken nextToken) throws (1: PrestoThriftServiceException ex1);
 
   /**
+   * Returns a batch of lookup splits.
+   * This method is called if index join is chosen in order to get splits for a given batch of keys.
+   * 
+   * @param schemaTableName schema and table name
+   * @param lookupColumnNames specifies columns and their order for keys
+   * @param outputColumnNames a list of column names to return
+   * @param keys keys for which records need to be returned
+   * @param outputConstraint constraint on the returned data
+   * @param maxSplitCount maximum number of splits to return
+   * @param nextToken token from a previous split batch or {@literal null} if it is the first call
+   * @return a batch of splits
+   */
+  PrestoThriftSplitBatch prestoGetLookupSplits(1:  PrestoThriftSchemaTableName schemaTableName, 2:  list<string> lookupColumnNames, 3:  list<string> outputColumnNames, 4:  PrestoThriftPageResult keys, 5:  PrestoThriftTupleDomain outputConstraint, 6:  i32 maxSplitCount, 7:  PrestoThriftNullableToken nextToken) throws (1: PrestoThriftServiceException ex1);
```

Comment:
Maybe call this `prestoGetIndexSplits`... I have found that most people are confused by the term `looup` in this context

Reply:
I'd substitute `lookup` for `index` through this code

Reply:
When I discussed it with Martin he used "lookup" as a term with an idea to change the way index join is implemented at some point. I'm fine with either name and can change it to "index" for consistency.

Reply:
Renamed a method to `getIndexSplits()`. However, I kept parameter name as `lookupColumnNames` as everywhere in presto code they are referred to a lookup, for example `lookupSchema` in `ConnectorIndexProvider`.

Reply:
I don't think we need to expose our internal implementation details (e.g., lookup) to this interface, so I would name everything index.

Reply:
Renamed swift parameter to `indexColumnNames`. Note, that SPI uses the term `lookupColumns`, so I kept this name in implementation.